the-collab-lab/tcl-12-smart-shopping-list
# Issues #12: Sort shopping list by how soon user will need to buy item again

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

- Ordered list items by calculated estimate of next time item would need to be purchased 
- Distinguished items that need to be purchased soon, kind of soon, not soon, inactive, and not yet purchased at the top by different colors

HAVE NOT COMPLETED: 
- ~~We struggled with the issue this week, mostly because our items are listed as objects vs an array in the Firestore database. We could not figure out how to order the items that have not been purchased alphabetically.~~
    - Terry recommended we move the "===" comparison to the top because it never got a chance to be read after the other conditionals
- ~~We also were trying to figure out how to move items that switch to inactive to the bottom of our item list. After researching for a few hours, we decided it would be best to ask the team and mentors if they could think of the best way to do this.~~
    - Terry again suggested code on how go about sorting Inactive before calculatedEstimate by repurposing the elapsedTime variable

## Related Issue

Issue #12 

## Acceptance Criteria



## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

<img width="862" alt="Screen Shot 2020-11-12 at 2 03 05 AM" src="https://user-images.githubusercontent.com/24923742/98912118-49ac1300-248b-11eb-8c9b-33d1a44e1028.png">


### After

<img width="865" alt="Screen Shot 2020-11-12 at 2 01 42 AM" src="https://user-images.githubusercontent.com/24923742/98912267-77915780-248b-11eb-81f9-e8a1b76421a5.png">


### After: Part 2

![After-2](https://user-images.githubusercontent.com/24923742/99114133-f5ea1880-25b5-11eb-9c8a-10806130dff1.png)


## Testing Steps / QA Criteria

- Switch to our branch (`git checkout gj-eb-sort-items`), then run `git pull origin gj-eb-sort-items`. Start server. 
- Input the token `mind shoo cuff`
- Open Firestore Console and find the corresponding token
- Items should appear in the app list in order of estimated days until next purchase (soonest, kind of soon, not soon, not purchased)
- Change the oldPurchased date in Firestore for an active item to a date that is more than calculatedEstimate x2 days before the lastPurchased date (if you need to, change calculatedEstimate to a lower number) to see it become inactive on the app
